src/python/pants/core/goals/test.py

```diff
@@ -140,6 +140,9 @@ def message(self) -> str:
             output = f"{output.rstrip()}\n\n"
         return f"{message}{output}"
 
+    def metadata(self) -> Optional[Dict[str, Any]]:
```

Comment:
Nit: while this return type isn't incorrect, you can change it to `Optional[]` to better reflect what's going on here.